docs/rfcs/component-universal-telemetry.md

```diff
@@ -0,0 +1,212 @@
+
+Two major challenges in pursuit of broadly normalized telemetry are (1) consistent attributes, and (2) automatic capture.
+
+This RFC represents an evolving consensus about the desired end state of component telemetry. It does _not_ claim
+to describe the final state of all component telemetry, but rather seeks to document some specific aspects. It proposes a set of
+attributes which are both necessary and sufficient to identify components and their instances. It also articulates one specific
+mechanism by which some telemetry can be automatically captured. Finally, it describes some specific metrics and logs which should
+be automatically captured for each kind of pipeline component.
+
+## Goals
+
+1. Define attributes that are (A) specific enough to describe individual component[_instances_](https://github.com/open-telemetry/opentelemetry-collector/issues/10534)
+   and (B) consistent enough for correlation across signals.
+2. Articulate a mechanism which enables us to _automatically_ capture telemetry from _all pipeline components_.
+3. Define specific metrics for each kind of pipeline component.
+4. Define specific logs for all kinds of pipeline component.
+
+## Attributes
+
+All signals should use the following attributes:
+
+### Receivers
+
+- `otel.component.kind`: `receiver`
+- `otel.component.id`: The component ID
+- `otel.signal`: `logs`, `metrics`, `traces`, `profiles`
+
+### Processors
+
+- `otel.component.kind`: `processor`
+- `otel.component.id`: The component ID
+- `otel.pipeline.id`: The pipeline ID
+- `otel.signal`: `logs`, `metrics`, `traces`, `profiles`
+
+### Exporters
+
+- `otel.component.kind`: `exporter`
+- `otel.component.id`: The component ID
+- `otel.signal`: `logs`, `metrics` `traces`, `profiles`
+
+### Connectors
+
+- `otel.component.kind`: `connector`
+- `otel.component.id`: The component ID
+- `otel.signal`: `logs`, `metrics` `traces`
+- `otel.output.signal`: `logs`, `metrics` `traces`, `profiles`
+
+Note: The `otel.signal`, `otel.output.signal`, or `otel.pipeline.id` attributes may be omitted if the corresponding component instances
+are unified by the component implementation. For example, the `otlp` receiver is a singleton, so its telemetry is not specific to a signal.
+Similarly, the `memory_limiter` processor is a singleton, so its telemetry is not specific to a pipeline.
+
+## Auto-Instrumentation Mechanism
+
+The mechanism of telemetry capture should be _external_ to components. Specifically, we should observe telemetry at each point where a
+component passes data to another component, and, at each point where a component consumes data from another component. In terms of the
+component graph, every _edge_ in the graph will have two layers of instrumentation - one for the producing component and one for the
+consuming component. Importantly, each layer generates telemetry ascribed to a single component instance, so by having two layers per
+edge we can describe both sides of each handoff independently.
```

Comment:
+1 for this mechanism. It should enable very useful debugging capabilities that are described in this [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34096) and were discussed in this [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/10963)